### PR TITLE
add phone_validator method

### PR DESCRIPTION
**Pull request by suryapratapsinghsuryavanshi**
### **Describe your change:**

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/phone_validator.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    True
+    >>> phone_validator("01234567896")
+    False
+    >>> phone_validator("919876543218")
+    True
+    >>> phone_validator("+91-1234567899")
+    False
+    """
+    # Created a regex pattern to produce a valid phone number
+    pat = re.compile(r"^(\+91[\-\s]?)?[0]?(91)?[789]\d{9}$")
```
Comment (mateuszz0000) [inline strings/phone_validator.py:19]: Why only +91 ? It's not very universal. Consider create function for each country.

File: strings/phone_validator.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    # Find their matches
+    match = re.search(pat, phone)
+    # if match are present
+    if match:
+        # the match is the same as the phone string it means they are valid phone number.
+        return match.string == phone
+    # if match are none so return False.
```
Comment (mateuszz0000) [inline strings/phone_validator.py:26]: All of these comments are redundant. Please remove

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]

File: strings/phone_validator.py (hunk 1 above)
Comment (suryapratapsinghsuryavanshi) [reply strings/phone_validator.py:19]: Ok, I change this function as a indian_phone_validator.

File: strings/phone_validator.py (hunk 2 above)
Comment (suryapratapsinghsuryavanshi) [reply strings/phone_validator.py]: ok, i removed them.

**Comment by ASHIKSAR** [comment]
I'm back

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]
Please remove all comments because code is simple

**Comment by mateuszz0000** [comment]
Tests are failing

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]
please take a look at the build and fix the issues

**Comment by suryapratapsinghsuryavanshi** [comment]
if __name__ == "__main__":
-    print(f'{indian_phone_validator("+918827897895")}\n')
\ No newline at end of file
+    print(f'{indian_phone_validator("+918827897895")}\n')
would reformat strings/indian_phone_validator.py
Oh no! 💥 💔 💥
1 file would be reformatted, 917 files would be left unchanged.

`How can I solve this problem`

**Comment by mateuszz0000** [comment]
Logs tell you everything! :) `No newline at end of file`

**Comment by mateuszz0000** [comment]
Please reformat code with `black`

**Review by mateuszz0000 — APPROVED** [review]

**Comment by suryapratapsinghsuryavanshi** [comment]
Thanks, this is my first contribution to the open-source world.

**Pull request merged**